AdobeStockImageAdminUi/Model/Listing/DataProvider.php

```diff
@@ -72,6 +74,21 @@ public function __construct(
         $this->searchResultFactory = $searchResultFactory;
     }
 
+    /**
+     * @inheritdoc
+     */
+    public function addFilter(\Magento\Framework\Api\Filter $filter)
+    {
+        if (self::FIELD_WORDS === $filter->getField()) {
+            $value = str_replace(['"', '\\'], '', $filter->getValue());
```

Comment:
I think the issue fixed by this PR is a client/api problem rather than UI. So, it would be better to fix it in Client module.

There is two possible ways to do this:
1. a new class implementing `Magento\AdobeStockClientApi\Api\SearchParameterProviderInterface` handling words filter can be created or 
2. If the provided escaping/replacing is applicable for all the filters: `Magento\AdobeStockClient\Model\SearchParametersProvider\SimpleFilters::apply` can be updated.

Reply:
@sivaschenko I've fixed it. You may check once again

Reply:
@IvanPletnyov @engcom-Golf thanks for update! Can you please cover the introduced parameters provided with a unit test with a data provider to outline how search query is modified by the replace and url encode.

Reply:
@sivaschenko unit tests was added.